X/XDiag/build_tarballs.jl

```diff
@@ -0,0 +1,92 @@
+uuid = Base.UUID("a83860b7-747b-57cf-bf1f-3e79990d037f")
+delete!(Pkg.Types.get_last_stdlibs(v"1.6.3"), uuid)
+
+
+name = "XDiag"
+version = v"0.2.0"
+
+include("../../L/libjulia/common.jl")
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/awietek/xdiag.git", "9a91ed0511299ea9f5ab9fbc1d5b8d2fd74a750d")
+]
+
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/xdiag
+
+Julia_PREFIX=${prefix}
+
+# Redefining BLAS_LAPACK symbols to use with OpenBLAS
+SYMB_DEFS=()
+for sym in sasum dasum snrm2 dnrm2 sdot ddot sgemv dgemv cgemv zgemv sgemm dgemm cgemm zgemm ssyrk dsyrk cherk zherk; do
+    SYMB_DEFS+=("-D${sym}=${sym}_64")
+done
+
+for sym in cgbcon cgbsv cgbsvx cgbtrf cgbtrs cgecon cgees cgeev cgeevx cgehrd cgels cgelsd cgemm cgemv cgeqp3 cgeqrf cgesdd cgesv cgesvd cgesvx cgetrf cgetri cgetrs cgges cggev cgtsv cgtsvx cheev cheevd cherk clangb clange clanhe clansy cpbtrf cpocon cposv cposvx cpotrf cpotri cpotrs cpstrf ctrcon ctrsyl ctrtri ctrtrs cungqr dasum ddot dgbcon dgbsv dgbsvx dgbtrf dgbtrs dgecon dgees dgeev dgeevx dgehrd dgels dgelsd dgemm dgemv dgeqp3 dgeqrf dgesdd dgesv dgesvd dgesvx dgetrf dgetri dgetrs dgges dggev dgtsv dgtsvx dlahqr dlangb dlange dlansy dlarnv dnrm2 dorgqr dpbtrf dpocon dposv dposvx dpotrf dpotri dpotrs dpstrf dstedc dsyev dsyevd dsyrk dtrcon dtrevc dtrsyl dtrtri dtrtrs ilaenv sasum sdot sgbcon sgbsv sgbsvx sgbtrf sgbtrs sgecon sgees sgeev sgeevx sgehrd sgels sgelsd sgemm sgemv sgeqrf sgeqp3 sgesdd sgesv sgesvd sgesvx sgetrf sgetri sgetrs sgges sggev sgtsv sgtsvx slahqr slangb slange slansy slarnv snrm2 sorgqr spbtrf spocon sposv sposvx spotrf spotri spotrs spstrf sstedc ssyev ssyevd ssyrk strcon strevc strsyl strtri strtrs zgbcon zgbsv zgbsvx zgbtrf zgbtrs zgecon zgees zgeev zgeevx zgehrd zgels zgelsd zgemm zgemv zgeqp3 zgeqrf zgesdd zgesv zgesvd zgesvx zgetrf zgetri zgetrs zgges zggev zgtsv zgtsvx zheev zheevd zherk zlangb zlange zlanhe zlansy zpbtrf zpocon zposv zposvx zpotrf zpotri zpotrs zpstrf ztrcon ztrsyl ztrtri ztrtrs zungqr; do
+    SYMB_DEFS+=("-D${sym}=${sym}_64")
+done
+
+export CXXFLAGS="${SYMB_DEFS[@]}"
+
+if [[ "${target}" == x86_64-apple-* ]]; then
+    # Needed to get std::visit working  
+    export MACOSX_DEPLOYMENT_TARGET=10.14
+fi
+
+cmake -S . \
+     -B build \
+    -D XDIAG_DISABLE_HDF5=On \
+    -DJulia_PREFIX=$Julia_PREFIX \
+    -DCMAKE_INSTALL_PREFIX=$prefix \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DXDIAG_JULIA_WRAPPER=On \
+    -DJlCxx_DIR=$prefix/lib/cmake \
+    -DBLAS_LIBRARIES=${libdir}/libopenblas64_.${dlext} \
+    -DLAPACK_LIBRARIES=${libdir}/libopenblas64_.${dlext}
+
+cmake --build build -j${nproc}
+cmake --install build
+
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = vcat(libjulia_platforms.(julia_versions)...)
```

Comment:
https://buildkite.com/julialang/yggdrasil/builds/12410#01913bd2-7bc1-4d3b-b483-054b65842b3e/676-3248
```
┌ Warning: /cache/build/yggy-amdci7-13/julialang/yggdrasil/X/XDiag/build/x86_64-linux-gnu-julia_version+1.8.2/hh7qNoLU/x86_64-linux-gnu-libgfortran5-cxx11-julia_version+1.8.2/destdir/lib/libxdiagjl.so contains std::string values!  This causes incompatibilities across the GCC 4/5 version boundary.  To remedy this, you must build a tarball for both GCC 4 and GCC 5.  To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:
│ 
│     platforms = expand_cxxstring_abis(platforms)
└ @ BinaryBuilder.Auditor /cache/julia-buildkite-plugin/depots/e2fd9734-29d8-45cd-b0eb-59f7104f3131/packages/BinaryBuilder/mzSV7/src/auditor/compiler_abi.jl:247
```
```suggestion
platforms = vcat(libjulia_platforms.(julia_versions)...)
platforms = expand_cxxstring_abis(platforms)
```